tests/candev/main.c

```diff
@@ -192,10 +208,16 @@ int main(void)
     puts("candev test application\n");
 
     isrpipe_init(&rxbuf, (uint8_t *)rx_ringbuf, sizeof(rx_ringbuf));
-#ifdef BOARD_NATIVE
+#if defined (BOARD_NATIVE)
     puts("Initializing Linux Can device");
     candev_linux_init( &linux_dev, &(candev_linux_conf[0]));    /* vcan0 */
     candev = (candev_t *)&linux_dev;
+
+#elif  defined(CAN_DRIVER_MCP2515)
+    puts("Initializing MCP2515");
+    candev_mcp2515_init(&mcp2515_dev, &mcp2515_conf);
```

Comment:
Probably I still did not fully understand the purpose of this test application. 
1. Shouldn't the initialization take place in `auto_init`? 
2. Why are the stm32 or esp32 peripherals not included?

Reply:
0. (about the usemodule for atmega, but I cannot comment on it there for some reason): I'm a bit confused on how you were able to still see that code in this PR? It's been removed a loooong time ago from the Makefile, and I can also not find the line anymore if I check it in "Files changed" of this PR? 
1. No. This test-app does not use auto_init. That was the whole purpose of the candev testapp: provide a simpler way of using CAN devices than the one that already existed in RIOT. This app still uses the candev abstraction, but does not use the can-dll anymore. 
1.1 (because I feel a follow up question coming): "But... isn't auto_init included when you use the `can` module in the makefile? "
--> Answer: no. We made auto_init explicit in 9b829c99a98f5b09c644c203b36f1462464f77ee. It is no longer implicitely activated. 
2. Because they are out of scope for this PR. This PR adds the driver for the MCP2515 device, and also adds the option to select this driver when using `tests/candev`. Other devices (such as the peripherals for stm32 and esp32) can also be added to this testapp, but I think it's best to do that in a separate PR.

Reply:
Ok, thanks. But lets use configurations defined in params. They are overwritable.